toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/internal/components/utilitynetwork/AddAssociationFromSourceViewModel.kt

```diff
@@ -174,12 +175,43 @@ internal class AddAssociationFromSourceViewModel(
      */
     val filteredFeatureCandidatesUiState: StateFlow<FeatureCandidatesUiState> = snapshotFlow {
         associatedFeaturesFilterText to _featureCandidatesUiState.value
-    }.map { (text, _) ->
+    }.map { (text, candidatesUiState) ->
         return@map if (text.isNotEmpty()) {
-            val filteredList = filterFeatureCandidates(text)
+            val candidateList = candidatesUiState.candidates.toMutableList()
+            var nextQueryParams = candidatesUiState.nextQueryParams
+            var filteredList = filterFeatureCandidates(text)
+            // If the filtered list is empty and there are more candidates to fetch, keep fetching
+            // until we either get results or run out of candidates
+            while (filteredList.isEmpty() && nextQueryParams != null) {
```

Comment:
should this logic be changed to 
```suggestion
            while (nextQueryParams != null) {
```
as we don't want to stop looking till `nextQueryParams` is not null.

Reply:
also can we just use `candidatesUiState.nextQueryParams` here instead of a local variable `nextQueryParams`
that way we will not have to update both the _featuerCandidateUiState and the nextQueryParams on line 198

Reply:
> as we don't want to stop looking till nextQueryParams is not null.

This is not true. We only want to fetch pages until we have some results. We do not want to fetch all the pages if there is no need to do so.

> also can we just use candidatesUiState.nextQueryParams here instead of a local variable nextQueryParams
that way we will not have to update both the _featuerCandidateUiState and the nextQueryParams on line 198

Since we're using the `map` operator , the `candidatesUiState` becomes outdated after we fetch a new page and update the `_featureCandidatesUiState` state with the new result. If we were to use `mapLatest` it would cancel the transformation and start a new one, but it is experimental. Hence we need to store an updated `nextQueryParams` for subsequent fetches.